### PR TITLE
Add link icon to release card header

### LINKED ISSUE #1273: Feature Request: be able to jump directly on a specific plugin release via an anchor
Currently there is no way to get a link pointing directly to a specific release.

Having something like an anchor on the release date for example would be really nice.

### PR DESCRIPTION
**Pull request by zbynek**
Fixes #1273 (the [remaining part](https://github.com/jenkins-infra/plugin-site/issues/1273#issuecomment-1462564927))

Also fixes some of the warnings Gatsby now produces in dev mode.

### CODE REVIEW

File: plugins/plugin-site/src/components/Pagination.jsx [hunk 1]
```diff
@@ -7,10 +7,6 @@ function Pagination({limit, page, pages, total, setPage}) {
     const [pagesToDisplay, setPagesToDisplay] = React.useState(5);
     const [marginPagesDisplayed, setMarginPagesDisplayed] = React.useState(2);
 
-    if (total == 0) {
```
Comment (zbynek) [inline plugins/plugin-site/src/components/Pagination.jsx:10]: The only place where Pagiation is called guards for `total == 0` already

**Comment by NotMyFault** [comment]
Looks good to me, but the build thinks otherwise

File: plugins/plugin-site/src/components/PluginReleases.jsx [hunk 2]
```diff
@@ -53,7 +53,7 @@ function PluginReleases({pluginId, versions}) {
                                         <a href={release.htmlURL} title="See the release on GitHub" className="github-icon d-flex"><ion-icon name="logo-github" /></a>
                                     </>)}
                                     <a className="anchor after" href={`#version_${version.version}`}>
-                                        <ion-icon name="link-outline"></ion-icon>
+                                        <ion-icon name="link-outline"/>
```
Comment (halkeye) [inline plugins/plugin-site/src/components/PluginReleases.jsx:56]: I believe web components must have closing tags, not self closing.

File: plugins/plugin-site/gatsby-browser.js [hunk 3]
```diff
@@ -59,7 +59,8 @@ exports.onClientEntry = function () {
                 /metrics\.itunes\.apple\.com\.edgesuite\.net\//i
             ]
         });
-        window.Sentry = Sentry;
-    });
+        return window.Sentry = Sentry;
+    /* eslint-disable no-console */
+    }).catch(e => console.log(e));
```
Comment (halkeye) [inline plugins/plugin-site/gatsby-browser.js]:
```suggestion
    }).catch(e => console.error(e));
```

File: plugins/plugin-site/src/components/PluginReleases.jsx (hunk 2 above)
Comment (zbynek) [reply plugins/plugin-site/src/components/PluginReleases.jsx:56]: You're right, self-closing custom elements are still not valid HTML (as far as I understand https://github.com/WICG/webcomponents/issues/624 ), but 
* ESlint prefers self-closing tags and we're using it this way for all icons in the repo
* This is .jsx so HTML parsing restrictions do not apply

**Review by NotMyFault — APPROVED** [review]

**Pull request merged**